lms/djangoapps/instructor/lti_connection.py

```diff
@@ -0,0 +1,140 @@
+"""
+Send grades to an OpenEdX LTI component
+"""
+import base64
+import hashlib
+import json
+import logging
+import urllib
+
+import mock
+from oauthlib.oauth1 import Client  # pylint: disable=F0401
+import requests
+
+from django.utils.translation import ugettext as _
+
+log = logging.getLogger(__name__)
+
+
+def validate_lti_passport(key, secret, url):
+    """
+    Sanity-check LTI passport credentials
+    """
+    resp = _send_lti_2_json_request('GET', url, key, secret)
+    if resp.status_code != 200:
+        raise LTIConnectionError(
+            _("LTI passport sanity check failed. Your lti_key ({key}) or lti_secret ({secret}) are probably incorrect.").format(
```

Comment:
This initially jumped out because we typically don't translate
exceptions. Then I saw that you're using it to bubble back an error
message.

Instead of bubbling an exception and treating the method as a `void`,
what do you think about returning a boolean success value from the
method and determine success/error message from there?

```python
def validate_lti_passport(key, secret, url):
    response = _send_lti_2_json_request('GET', url, key, secret)
    return response.status_code == 200
# ...
def update_grades(self, grade_data):
    # ...
    credentials_are_valid = lti_connection.validate_lti_passport(self.key, self.secret, test_url)
    if not credentials_are_valid:
        message = _(
            "LTI passport sanity check failed. Your lti_key ({key}) or "
            "lti_secret ({secret}) are probably incorrect."
        ).format(
            key=self.key,
            secret=self.secret,
        )
        # ...
```